app/workers/publishing_api_withdrawal_worker.rb

```diff
@@ -1,24 +1,40 @@
 class PublishingApiWithdrawalWorker < PublishingApiWorker
-  def perform(content_id, explanation, locale, allow_draft = false)
+  # `explanation` and `unpublished_at` come from the unpublishing object. Rather than
+  # performing a database query here to look up the `unpublishing` linked to the most
+  # recent edition, we pass it in directly because the `unpublishing` isn't always
+  # saved in the database yet when this worker runs.
+  def perform(content_id, explanation, locale, allow_draft = false, unpublished_at = nil)
     check_if_locked_document(content_id: content_id)
 
-    unpublished_at = Edition
-      .joins(:document)
-      .where(documents: { content_id: content_id })
-      .where(state: "withdrawn")
-      .pluck(:updated_at)
-      .first
-
     Services.publishing_api.unpublish(
       content_id,
       type: "withdrawal",
       locale: locale,
       explanation: Whitehall::GovspeakRenderer.new.govspeak_to_html(explanation),
       allow_draft: allow_draft,
-      unpublished_at: unpublished_at,
+      unpublished_at: find_unpublished_at(content_id, unpublished_at),
     )
   rescue GdsApi::HTTPNotFound, GdsApi::HTTPUnprocessableEntity
     # nothing to do here as we can't unpublish something that doesn't exist
     nil
   end
+
+private
+
+  def find_unpublished_at(content_id, given_unpublishing_at)
+    if given_unpublishing_at
+      # We call this job both directly and via Sidekiq. When called by Sidekiq, the date gets turned
+      # into a string (because jobs must be JSON serialisable) and `Services.publishing_api.unpublish`
+      # rejects it.
+      given_unpublishing_at.respond_to?(:utc) ? given_unpublishing_at : Time.zone.parse(given_unpublishing_at)
```

Comment:
Minor: we could make this slightly neater with `Time.zone.parse(given_unpublishing_at.to_s)`. Also, I think you mean `unpublish*ed*_at`.